src/sync/app.cpp

```diff
@@ -73,10 +73,16 @@ App::App(const Config& config)
 
 static Optional<AppError> check_for_errors(const Response& response)
 {
+    bool http_status_code_is_fatal = response.http_status_code >= 300 ||
+        (response.http_status_code < 200 && response.http_status_code != 0);
+    
     try {
-        if (auto ct = response.headers.find("Content-Type"); ct != response.headers.end() && ct->second == "application/json") {
+        auto ct = response.headers.find("Content-Type");
+        if (ct != response.headers.end() && ct->second == "application/json" && http_status_code_is_fatal) {
```

Comment:
This change is causing the other integration tests to fail. It is possible for stitch to return an error code with http 200, for example when registering an email account twice the request is processed but denied. I will change it back in https://github.com/realm/realm-object-store/pull/941 but if there's something I'm missing about this let me know